server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -0,0 +1,97 @@
+package services.ti;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import forms.UpdateApplicantDob;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+import java.util.Optional;
+import javax.inject.Inject;
+import models.Account;
+import models.Applicant;
+import models.TrustedIntermediaryGroup;
+import play.data.Form;
+import repository.SearchParameters;
+import repository.UserRepository;
+import services.DateConverter;
+import services.applicant.exception.ApplicantNotFoundException;
+
+public class TrustedIntermediaryService {
+
+  private final DateConverter dateConverter;
+  private final UserRepository userRepository;
+
+  @Inject
+  public TrustedIntermediaryService(DateConverter dateConverter, UserRepository userRepository) {
+    this.dateConverter = Preconditions.checkNotNull(dateConverter);
+    this.userRepository = Preconditions.checkNotNull(userRepository);
+  }
+
+  public ImmutableList<Account> retriveSearchResult(
+      SearchParameters searchParameters,
+      Optional<TrustedIntermediaryGroup> trustedIntermediaryGroup) {
+    return filterAccountsBySearchParams(searchParameters, trustedIntermediaryGroup.get());
+  }
+
+  public ImmutableList<Account> filterAccountsBySearchParams(
+      SearchParameters searchParameters, TrustedIntermediaryGroup tiGroup) {
+    ImmutableList<Account> allAccounts = tiGroup.getManagedAccounts();
+    allAccounts =
```

Comment:
Not blocking: Saving this off into a separate variable isn't really saving any lines of code and the majority of the logic is already indented. I'd suggest just returning the result of the stream directly rather than saving it.

Concretely:
```
return tiGroup.getManagedAccounts().stream()
  .filter(...)
  .collect(...);
```